src/cmake/dependency_utils.cmake

```diff
@@ -0,0 +1,422 @@
+# Copyright Contributors to the OpenImageIO project.
+# SPDX-License-Identifier: Apache-2.0
+# https://github.com/AcademySoftwareFoundation/OpenImageIO
+
+
+set_cache (${PROJECT_NAME}_REQUIRED_DEPS ""
+           "Additional dependencies to consider required (semicolon-separated list, or ALL)")
+set_cache (${PROJECT_NAME}_OPTIONAL_DEPS ""
+           "Additional dependencies to consider optional (semicolon-separated list, or ALL)")
+set_option (${PROJECT_NAME}_ALWAYS_PREFER_CONFIG
+            "Prefer a dependency's exported config file if it's available" OFF)
+set_cache (${PROJECT_NAME}_DEPENDENCY_BUILD_TYPE "Release"
```

Comment:
Ah, I see. I was thinking that mostly people would want the dependencies to be optimized even if they were building a debug OIIO. But now I see that this my Unix bias speaking -- I think maybe that in Windows land, debug and optimized builds are not freely mixable?

Maybe it's better if we initialize the build mode of dependencies to default to that of the present OIIO build. (But still allow override.)

Reply:
I mix debug and release all the time on Windows. The thing to keep in mind is that there are various types of debugging flags. It's the mixing of the runtime that is a problem. Making a debug build with no optimization and including symbols (what we normally do on unix) won't cause issues.
The frustrating part is that you need to make sure that all statically linked dependencies follow the same convention and most projects default to the convention that "debug" builds also use the debug runtime.

So one fix might be to modify the OIIO cmake so that debug builds don't link to the debug runtime. That is fine for shared libraries of OIIO. But if OIIO is used as a static lib, the consumer of OIIO will now also be required to make the same change and I don't think OIIO should impose this.

We could add a cmake option to pick whether debug builds should use debug runtimes and have it default to yes so by default this works as most users would expect. However, if you're the type of user that uses static libs and knows how to make debug builds that don't use debug runtimes, it's not that hard to just run ccmake and change the "release" mode to use /Od and other debug flags. 

So all this to say I agree with lg's above proposal.

Reply:
I pushed an update that changes the default build mode for dependencies to match that of the main project.

The different Windows debugging incompatibilities are a bit over my head, so I'll leave it to somebody who knows that platform better to come back and enhance this with more flexibility later.